pkg/cloud/azure/actuators/machine_scope.go

```diff
@@ -398,3 +413,32 @@ func getEnvironment(m *MachineScope) (*azure.Environment, error) {
 
 	return &env, nil
 }
+
+// getTagList is for getting the merged tag list present in
+// AzureMachineProviderSpec and Infrastructure.Status. Both
+// will have the same tags, unless AzureMachineProviderSpec.Tags
+// has been modified to add/update/delete any tag. Merge will be
+// necessary in delete case, to honour the user-defined tags
+// in Infrastructure.Status and those to newly created resources.
+func getTagList(machineSpecTags, infraStatusTags map[string]string) map[string]*string {
+	if machineSpecTags == nil && infraStatusTags == nil {
+		return nil
+	}
+
+	// infraStatusTags will not be empty when GetCloudEnvironment()
+	// is used for fetching tags in Infrastructure.Status, but
+	// keeping the check for otherwise.
+	if infraStatusTags == nil {
+		infraStatusTags = make(map[string]string)
+	}
+
+	// merge tags present in Infrastructure.Status with
+	// the tags configured in AzureMachineProviderSpec, with
+	// precedence given to those in AzureMachineProviderSpec
+	// for new or updated tags.
+	for k, v := range machineSpecTags {
+		infraStatusTags[k] = v
+	}
+
+	return *(to.StringMapPtr(infraStatusTags))
```

Comment:
What if merged tag list has more than 10 tags?

Reply:
I guess we have to return an error and ask the user to remove some of the additional tags, are any of the tags being set on the infra/machines part of the openshift owned tags? Eg the cluster ID? Because if they are, we need to exclude those from the count as well

IIRC though, Machines can have more than 15 tags right? So perhaps this isn't an issue? The user has their up to 10 tags in the infra and then we can top that up with Machine tags specifically to whatever the limit happens to be for a VM in azure

Reply:
Yes, all the Azure resource created by the operator has max limit of 50 tags. Reserving 5 for openshift, a user can add 45 tags.
I will add a check to limit the user defined tags to not more than 45?